app/models/mixins/relationship_mixin.rb

```diff
@@ -565,11 +618,14 @@ def add_children(*child_objs)
   alias_method :add_child, :add_children
 
   def parent=(parent)
+    return call_ancestry_method(:parent=, parent) if use_ancestry?
+
     if parent.nil?
       remove_all_parents
     else
       parent.with_relationship_type(relationship_type) do
         parent_rel = parent.init_relationship
+        save!
```

Comment:
Is this still necessary?

did this get uncovered with your tests?

Reply:
sorry but I have to be stubborn about this one too. it's necessary to fix the tests that fail otherwise with the active record validation:

```
 15) ManageIQ::Providers::Inventory::Persister check we can load stack resource records from the DB finds in one batch after the scanning
      Failure/Error:
        all_relationships.create!(
          :relationship => (parent_rel.nil? ? relationship_type : parent_rel.relationship),
          :parent       => parent_rel

        )
      ActiveRecord::RecordNotSaved:
        You cannot call create unless the parent is saved

      # ./app/models/mixins/relationship_mixin.rb:499:in `add_relationship'
      # ./app/models/mixins/relationship_mixin.rb:511:in `init_relationship'
      # ./app/models/mixins/relationship_mixin.rb:628:in `block in parent='
      # ./app/models/mixins/relationship_mixin.rb:89:in `with_relationship_type'
...
```

Reply:
So I'm pretty sure that testing this with an anonymous class is going to necessitate this `save!` and you can fight me about it later.